port/win/env_win.cc

```diff
@@ -1637,6 +1637,7 @@ class WinEnv : public Env {
     return s;
   }
 
+#if (_WIN32_WINNT >= _WIN32_WINNT_WIN8)
```

Comment:
If I was doing this, I'd probably do something like the following, so that the main code is not ifdef'd, but if win7 compat is enabled, detect at runtime if the preferred API is available or not.  the overhead should be low and only paid for by the people that need rocksdb on win7.

the following is completely untested and undoubtably broken.  but again, it's approximately what i'd do if i needed rocksdb working on win7

```c++
#ifdef WIN7_COMPAT
static void GetSystemTimePreciseAsFileTime(LPFILETIME lpSystemTimeAsFileTime)
{
    // assumes VC does thread-safe init of local statics, otherwise,
    // complexify this code with a suitable lock
    static auto p = GetProcAddress(GetModuleHandle("kernel32.dll"),
            "GetSystemTimePreciseAsFileTime"));
    if (p) {
        return ((void WINAPI (*)(LPFILETIME))p)(lpSystemTimeAsFileTime);
    } else {
        LARGE_INTEGER li;
        QueryPerformanceCounter(&li);
        // pseudo code
        *lpSystemTimeAsFileTime = scaled(li);
    }
}
#endif
```